sdk/core/Azure.Core/src/RehydrationToken.Serialization.cs

```diff
@@ -105,7 +105,7 @@ void IJsonModel<RehydrationToken>.Write(Utf8JsonWriter writer, ModelReaderWriter
             writer.WritePropertyName("version"u8);
             writer.WriteStringValue(Version);
             writer.WritePropertyName("headerSource"u8);
-            writer.WriteStringValue(HeaderSource.ToString());
+            writer.WriteStringValue(HeaderSource?.ToString());
```

Comment:
Hasn't this type has publicly GA'ed?  Is this a bug fix, or a behavioral change?  We should avoid behavioral changes to publicly GA'ed types.

Reply:
Unfortunately this is a bug fix, found it during this PR change.

Reply:
Created https://github.com/Azure/azure-sdk-for-net/pull/42662 for this fix.

Reply:
Fix is merged